### ISSUE TITLE
Enable non-indefinite pins

### ISSUE THREAD

**Issue by lanzafame** [ipfs/ipfs-cluster#948]
**Describe the feature you are proposing**

Allow for a duration to be passed with a pin request, and when that duration is complete the pin is unpinned automatically by cluster.

**Comment by kishansagathiya** [comment]
Can you add some more details? This sounds like pin with expiration.

**Comment by lanzafame** [comment]
>  This sounds like pin with expiration
yes, is there an issue open for that? I looked but couldn't find anything.

To clarify though, the semantics are slightly different, `expiry` in my head, suggests that the user is passing in the date+time that they want the pin to unpin. Whereas, again in my head, `duration` is simply a length of time from when it is pinned.

so in api sense
expiry:
```
POST /api/pin?hash=<hash>&expiry=20200101000100
```
duration:
```
POST /api/pin?hash=<hash>&duration=120h
```

Now, after writing that out, both are probably really useful.

**Comment by kishansagathiya** [comment]
Both of these are supported, `expiry` as `expire-at` (it is time.MarshalText()), and `duration` as `expire-in`

**Comment by kishansagathiya** [comment]
Through cmd, only `expire-in` is supported.

**Comment by hsanjuan** [comment]
hey, this seems fixed by #923

**Comment by lanzafame** [comment]
Ahh wonderful, I was simply searching for the wrong thing when I went looking. Sorry for the bother, and thanks for implementing it before I even asked :grinning:
